twisted/test/test_task.py

```diff
@@ -1199,3 +1199,197 @@ def main(reactor):
         exitError = self.assertRaises(
             SystemExit, task.react, main, [], _reactor=r)
         self.assertEqual(0, exitError.code)
+
+
+
+class _DummyException(Exception):
+    """
+    Exception type to be used in testing.
+    """
+
+
+
+class TimeoutDeferredTests(unittest.SynchronousTestCase):
+    """
+    Tests for the function L{task.timeoutDeferred}
+    """
+    def test_propagatesSuccessResultBeforeTimeoutDefaultTranslation(self):
+        """
+        The L{defer.Deferred} callbacks with the result if it succeeds before
+        the timeout (e.g. timing out the L{defer.Deferred} does not obscure
+        the callback value) when using the default translation function.
+        """
+        clock = Clock()
+        d = defer.Deferred()
+        task.timeoutDeferred(d, 10, clock)
+        d.callback("Result")
+        self.assertEqual(self.successResultOf(d), "Result")
+
+        # The timeout never happens - no errback occurs
+        clock.advance(15)
+        self.assertIsNone(self.successResultOf(d))
```

Comment:
this is confusing... I understand how self.successResultOf works and why we got None here, but this looks unrelated to this test.... 

so maybe this should just check that the result is not a Failure


if later successResultOf is somehow updated to preserver the result, this test will fail but no good reason :)